File: my_cpp_pkg/src/add_two_ints_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);                         
    auto node = std::make_shared<rclcpp::Node>("add_two_ints_client");   
    auto client = node->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints"); 

    // Wait for the service to become available
    while (!client->wait_for_service(std::chrono::seconds(1))) {
        RCLCPP_WARN(node->get_logger(), "Service not available, waiting again...");
    }

    // Create a request
    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = 3;
    request->b = 8;

    // Send the request asynchronously
    auto future = client->async_send_request(request);

    // Wait for the response
    if (rclcpp::spin_until_future_complete(node, future) == rclcpp::FutureReturnCode::SUCCESS) {
        auto response = future.get();
        RCLCPP_INFO(node->get_logger(), "%ld + %ld = %ld", request->a, request->b, response->sum);
    } else {
        RCLCPP_ERROR(node->get_logger(), "Failed to call service add_two_ints");
    }

    rclcpp::shutdown();                               
    return 0;
}

File: turtlesim_catch_all/my_python_pkg/my_python_pkg/add_two_ints_client.py
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

def main(args=None):
    rclpy.init(args=args)
    node = Node('add_two_ints_client')

    # Create a service client
    client = node.create_client(AddTwoInts, 'add_two_ints')

    # Wait for the service to be available
    while not client.wait_for_service(timeout_sec=1.0):
        node.get_logger().warn('Service not available, waiting for the server...')

    # Prepare the request
    request = AddTwoInts.Request()
    request.a = 10
    request.b = 20

    # Call the service asynchronously
    future = client.call_async(request)
    rclpy.spin_until_future_complete(node, future)

    # Handle the response
    try:
        response = future.result()
        node.get_logger().info(f"Response: a = {request.a}, b = {request.b}, sum = {response.sum}")
    except Exception as e:
        node.get_logger().error(f"Service call failed: {str(e)}")

    # Cleanup
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
